project-koku/koku
# [COST-4268] Do not attempt creating a new cluster without cluster_id

## Jira Ticket

[COST-4268](https://issues.redhat.com/browse/COST-4268)

## Description

This change add a check to skip creating a new entry in `reporting_ocp_clusters` tables if we don't have a cluster_id.

## Testing

1. Checkout Branch
2. Restart Koku
3. Load test data
5. launch shell with `make shell` and run the snippet below
```py
In [1]: from api.provider.models import Provider

In [2]: from masu.database.report_manifest_db_accessor import ReportManifestDBAccessor

In [3]: from masu.processor.ocp.ocp_report_parquet_summary_updater import OCPReportParquetSummaryUpdater

In [4]: manifest_id = 1

In [5]: with ReportManifestDBAccessor() as manifest_accessor:
   ...:     manifest = manifest_accessor.get_manifest_by_id(manifest_id)
   ...: 

```
i. try a valid OCP provider:
```py

In [6]: ocp_provider = Provider.objects.filter(type=Provider.PROVIDER_OCP, authentication__credentials__cluster_id="my-ocp-cluster-1").first()

In [7]: updater = OCPReportParquetSummaryUpdater("org1234567", ocp_provider, manifest)

```

- This should run fine and you should see cluster_id and cluster_alias in logs
```py
[2023-10-06 19:36:11,460] INFO None 20409 found cluster_id: my-ocp-cluster-1 for provider-uuid: f88c5ab6-68b2-4097-b31a-9f7e57ed1b7c
[2023-10-06 19:36:11,463] INFO None 20409 found cluster_alias: Test OCP on AWS for cluster-id: my-ocp-cluster-1
```

ii. try an invalid OCP provider:
```py
In [10]: aws_provider = Provider.objects.filter(type=Provider.PROVIDER_AWS_LOCAL).first()
In [11]: updater = OCPReportParquetSummaryUpdater("org1234567", aws_provider, manifest)
```

- A value error should be raised:
```py
[2023-10-06 19:37:32,863] INFO None 20409 found cluster_id: None for provider-uuid: 3c868e47-85e5-4864-8528-b20e77facbe6
[2023-10-06 19:37:32,863] ERROR None 20409 {'message': 'Missing cluster_id for provider: 3c868e47-85e5-4864-8528-b20e77facbe6', 'tracing_id': '', 'provider_uuid': '3c868e47-85e5-4864-8528-b20e77facbe6'}
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[11], line 1
----> 1 updater = OCPReportParquetSummaryUpdater("org1234567", aws_provider, manifest)

File ~/Documents/redhat/cost-management/Project-koku/koku/koku/masu/processor/ocp/ocp_report_parquet_summary_updater.py:49, in OCPReportParquetSummaryUpdater.__init__(self, schema, provider, manifest)
     42     msg = f"Missing cluster_id for provider: {self._provider.uuid}"
     43     LOG.error(
     44         log_json(
     45             msg=msg,
     46             provider_uuid=provider.uuid,
     47         )
     48     )
---> 49     raise ValueError(msg)
     51 self._cluster_alias = get_cluster_alias_from_cluster_id(self._cluster_id)
     52 self._date_accessor = DateAccessor()

ValueError: Missing cluster_id for provider: 3c868e47-85e5-4864-8528-b20e77facbe6
```

## Notes

...
